app/components/Login.react.js

```diff
@@ -52,25 +54,62 @@ class Login extends Component {
         this.buildOauthUrl = this.buildOauthUrl.bind(this);
         this.oauthPopUp = this.oauthPopUp.bind(this);
         this.logIn = this.logIn.bind(this);
-        this.checkAnyUserAccountIsSaved = this.checkAnyUserAccountIsSaved.bind(this);
     }
 
-    componentWillReceiveProps(nextProps) {
-        this.checkAnyUserAccountIsSaved().then(isSaved => {
-            if (isSaved) {
-                window.location.assign(connectorUrl);
-            }
-        });
+    componentDidMount() {
+        const {serverType} = this.state;
+        const that = this;
+        const interval = setInterval(() => {
+              usernameLogged = cookie.load('db-connector-user');
+              if (usernameLogged) {
+                  if (serverType === ONPREM) {
+                      that.setState({
+                          status: 'authorized',
+                          statusMessage: 'Saving user information...'
+                      });
+                      that.saveDomainToSettings().then(res => {
+                          if ( res.status === 200 ) {
+                              window.location.assign(connectorUrl);
+                          } else {
+                              that.setState(
+                                  status: 'failure',
+                                  statusMessage: 'An error occured. Please try again.'
+                              )
+                          }
+                      });
+                  }
+                  else {
+                      window.location.assign(connectorUrl);
+                  }
+              }
+        }, 1000);
+
     }
 
-    componentDidMount() {
-        this.checkAnyUserAccountIsSaved().then(isSaved => {
-            if (isSaved) {
-                window.location.assign(connectorUrl);
-            }
+    saveDomainToSettings() {
+
+        const {domain} = this.state;
+        let PLOTLY_API_SSL_ENABLED = true;
+        let PLOTLY_API_DOMAIN = '';
+        if (domain.startsWith('https://')) {
+            PLOTLY_API_SSL_ENABLED = true;
+            PLOTLY_API_DOMAIN = domain.replace('https://', '');
+        } else {
+            PLOTLY_API_SSL_ENABLED = false;
+            PLOTLY_API_DOMAIN = domain.replace('http://', '');
+        }
+        return fetch(`${baseUrlWrapped}/settings`, {
+            method: 'PATCH',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                PLOTLY_API_DOMAIN, PLOTLY_API_SSL_ENABLED
+            })
```

Comment:
why is this needed? for on-prem, these settings will get set in run-time by the docker variables. for users running it themselves, they'll just use the default public cloud settings or modify the settings.yml file directly

Reply:
ah, i see that this was done below. Hm... was there a good reason for it? I can't think of any right now.

Reply:
I was not sure, so I kept it instead of deleting it. I am okay with removing it though.. 😸

Reply:
As discussed offline, we are leaving it as it is for now ..